### ISSUE TITLE
Improve MenuButton accessibility to follow WAI-ARIA Menu Button pattern

### ISSUE THREAD

**Issue by vanessatran-ddi** [GovAlta/ui-components#3452]
Reference:
https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/
The current `MenuButton` component is missing several WAI-ARIA attributes, Screen reader support is incomplete.

## Acceptance Criteria:
1. Add `aria-haspopup="true"` when we trigger the button: so screen reader can announce that activating the button to open the menu.
2. Add `aria-expanded="true/fase"` announces whether the button is open or not.
3. Add `role="menu"` identifies the popup as a menu to screen readers
4. `role="menu-item"` for each `MenuAction` will identifies each menu action  as a menu item.
5. `aria-controls` to associate the button with the menu it controls
